tests/rundb/test_httpdb.py

```diff
@@ -323,10 +326,92 @@ def test_bearer_auth(create_server):
     with pytest.raises(mlrun.errors.MLRunUnauthorizedError):
         db.list_runs()
 
-    db.token = token
+    db.token_provider = StaticTokenProvider(token)
     db.list_runs()
 
 
+def test_client_id_auth(requests_mock: requests_mock_package.Mocker):
+    """
+    Test the httpdb behavior when using a client-id OAuth token. Test verifies that:
+    - Token is retrieved successfully, and kept in the httpdb class.
+    - Token is added as Bearer token when issuing API calls to BE.
+    - Token is refreshed when its expiry time is nearing.
+    - Some error flows when token cannot be retrieved - such as that token is still used while it hasn't expired.
+    """
+
+    token_url = "https://mock/token_endpoint/protocol/openid-connect/token"
+    test_env = {
+        "MLRUN_AUTH_TOKEN_ENDPOINT": token_url,
+        "MLRUN_AUTH_CLIENT_ID": "some-client-id",
+        "MLRUN_AUTH_CLIENT_SECRET": "some-client-secret",
+    }
+
+    mlrun.mlconf.auth_with_client_id = True
+    os.environ.update(test_env)
```

Comment:
In the same spirit of Gal's comment, I suggest using pytest's  `monkeypatch` fixture:
https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.setenv
Proper teardown guaranteed 😉

Reply:
I actually looked for ways to do that and `monkeypatch` was an option I considered - I seem to recall seeing some discussions talking about how `monkeypatch` persists beyond a single test and causing "interesting" issues. I'll check again and if it works I'll indeed use it.

Reply:
Anyway, decided to use `monkeypatch` - seems ok.